internal/serve/httphandler/mfa_handler.go

```diff
@@ -43,8 +46,35 @@ func (h MFAHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	deviceID := req.Header.Get(DeviceIDHeader)
+	if deviceID == "" {
+		httperror.BadRequest("Device-ID header is required", nil, nil).Render(rw)
+		return
+	}
+
+	if reqBody.MFACode == "" {
+		extras := map[string]interface{}{"mfa_code": "MFA Code is required"}
+		httperror.BadRequest("Request invalid", nil, extras).Render(rw)
+		return
+	}
+
+	user, err := h.AuthManager.GetUserByDeviceID(ctx, strings.TrimSpace(deviceID))
```

Comment:
🚨 It doesn't look like the implementation of `GetUserByDeviceID` is returning the user roles.